qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -60,7 +62,9 @@ class UnitarySynthesis(TransformationPass):
 
     def __init__(self,
                  basis_gates: List[str],
-                 approximation_degree: float = 1):
+                 approximation_degree: float = 1,
+                 coupling_map = None,
```

Comment:
Would it be better to pass a `PassManagerConfig` to the plugin to avoid extending parameters further and further (for example, in case of a potential noise-aware synthesis)? Maybe `approximation_degree` should be part of `PassManagerConfig`?

Reply:
The issue with using a `PassManagerConfig` is that it is only used by transpile and the preset pass managers? If a user wants to create a custom pass manager or directly call this pass they'll have to go an instantiate this config object to pass to the pass constructor which nothing else requires. For the preset pass manager case I think it makes things simpler but I think we have to consider the standalone case too.

The one thing I thought about doing was making the api free form and just have the abstract `run()` signature take all kwargs and then the plugins can use them opportunistically so that gives us a bit more flexibility in the api (as we can add parameters over time). But I decided on being explicit like this so we can make the api a bit more well defined.